common/testing/runtime/goroutine.go

```diff
@@ -70,63 +72,89 @@ func WithNumGoRoutines(numGoRoutines int) func(*WaitOptions) {
 	}
 }
 
-// WaitGoRoutineWithFn waits for a go routine with the given function to appear within the duration.
-func WaitGoRoutineWithFn(t testing.TB, fn any, opts ...func(*WaitOptions)) {
-	t.Helper()
-
+// WaitGoRoutineWithFn waits for a go routine with the given function to appear in call stacks,
+// using different WaitOptions, and returns the number of attempts needed to find the go routine.
+func WaitGoRoutineWithFn(t testing.TB, fn any, opts ...func(*WaitOptions)) int {
```

Comment:
This is alright for now, but we should talk about a less magical way to accomplish that.